scripts/dts/edtlib.py

```diff
@@ -1597,16 +1597,11 @@ def _check_prop_type_and_default(prop_name, prop_type, required, default,
              "like '#pwm-cells' and 'pwm-names' get derived from 'pwms', for "
              "example.".format(prop_name, binding_path))
 
-    # Check default
+    # Check default, but not if required
 
-    if default is None:
+    if (default is None) or required:
         return
 
-    if required:
-        _err("'default:' for '{}' in 'properties:' in {} is meaningless in "
-             "combination with 'required: true'"
-             .format(prop_name, binding_path))
```

Comment:
Might be better to just remove this check and still let the checks below run when there's a default.

Reply:
Could you explain that?  If the property is required the default should be superseded by the value provided in the devicetree source, so there's no point continuing to validate the default.  In particular, it might not be compatible with the type of the property if that changed when the property was made `required` in an override.

Reply:
Nice to flag stuff like this, as it probably wasn't intended:

```
foo:
    required: true
    type: int
    default: "cool string that doesn't make sense"
```

Reply:
It would be, but I don't know how to do that, and I'm not going to fuss with it.

I've opened #21181 for this.  This PR will be blocked until that's resolved.  If the decision is that we have to update the SOC bindings for every target platform, then that's what we'll do.

Reply:
Would just need to do this I think, removing all other changes:

```diff
diff --git a/scripts/dts/edtlib.py b/scripts/dts/edtlib.py
index 9877f1157e..84532fca54 100644
--- a/scripts/dts/edtlib.py
+++ b/scripts/dts/edtlib.py
@@ -1604,11 +1604,6 @@ def _check_prop_type_and_default(prop_name, prop_type, required, default,
     if default is None:
         return
 
-    if required:
-        _err("'default:' for '{}' in 'properties:' in {} is meaningless in "
-             "combination with 'required: true'"
-             .format(prop_name, binding_path))
-
     if prop_type in {"boolean", "compound", "phandle", "phandles",
                      "phandle-array"}:
         _err("'default:' can't be combined with 'type: {}' for '{}' in "
```

Reply:
Ah; sorry, I thought you were giving an example where the property type changed in an override, which was a case I was concerned about.  OK, I've just removed the check completely.